src/nomad_material_processing/general/schema.py

```diff
@@ -0,0 +1,278 @@
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.NumberEditQuantity, defaultDisplayUnit='minute'
+        ),
+        unit='second',
+    )
+    temperature = Quantity(
+        type=np.float64,
+        description='The temperature of the etching process.',
+        a_eln={'component': 'NumberEditQuantity', 'defaultDisplayUnit': 'celsius'},
+        unit='celsius',
+    )
+    agitation = Quantity(
+        type=MEnum(
+            'Magnetic Stirring',
+            'Sonication',
+        ),
+        description='The agitation method used during the etching process.',
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.EnumEditQuantity,
+        ),
+    )
+    etching_reagents = SubSection(section_def=CompositeSystem, repeats=True)
+
+
+class Etching(Process, EntryData):
+    """
+    Class autogenerated from yaml schema.
+    """
+
+    m_def = Section(
+        a_eln=None,
+    )
+    recipe = Quantity(
+        type=Recipe,
+        description=""" The recipe used for the process. If a recipe is found,
+           all the data is copied from the Recipe within the Process.
+           """,
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.ReferenceEditQuantity,
+        ),
+    )
+    steps = SubSection(
+        description="""
+        The steps of the etching process.
+        """,
+        section_def=EtchingStep,
+        repeats=True,
+    )
+
+
+class EtchingRecipe(Etching, Recipe, EntryData):
+    """
+    A Recipe for an etching process.
+    """
+
+    m_def = Section(
+        a_eln={'hide': ['datetime', 'samples']},
+    )
+
+
+class Annealing(Process, EntryData):
+    """
+    Class autogenerated from yaml schema.
+    """
+
+    m_def = Section(
+        a_eln={'hide': ['steps']},
+    )
+    recipe = Quantity(
+        type=Recipe,
+        description=""" The recipe used for the process. If a recipe is found,
+           all the data is copied from the Recipe within the Process.
+           """,
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.ReferenceEditQuantity,
+        ),
+    )
+    temperature = Quantity(
+        type=np.float64,
+        description='The temperature of the annealing process.',
+        a_eln={'component': 'NumberEditQuantity', 'defaultDisplayUnit': 'celsius'},
+        unit='celsius',
+    )
+    duration = Quantity(
+        type=np.float64,
+        description='The elapsed time since the annealing process started.',
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.NumberEditQuantity, defaultDisplayUnit='minute'
+        ),
+        unit='second',
+    )
+    annealing_reagents = SubSection(
+        section_def=CompositeSystemReference,
+    )
+
+
+class AnnealingRecipe(Annealing, Recipe, EntryData):
+    """
+    A Recipe for an annealing process.
+    """
+
+    m_def = Section(
+        a_eln={'hide': ['datetime', 'samples']},
+    )
+
+
+class Cleaning(Process, EntryData):
```

Comment:
Again, I would add steps and a description.

Reply:
added!